### PR TITLE
Some progressbar dialog fixes

### PR DESCRIPTION
**Pull request by bigworld12**
i think this is how progressbar dialog should be , removed the obsolete from IsOpen property and made it read only , also added cancelled event

### CODE REVIEW

File: MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs [hunk 1]
```diff
@@ -12,23 +12,36 @@ public class ProgressDialogController
...
+        /// Gets if the Cancel button has been pressed.
+        /// </summary>        
+        public bool IsCancelled {get{ return m_iscanceled;}}
+        private bool m_iscanceled;
+
+        /// <summary>
+        /// Gets if the wrapped ProgressDialog is open.
+        /// </summary>        
+        public bool IsOpen { get { return m_isopen; } }
+        private bool m_isopen;
```
Comment (punker76) [inline MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs]: why this change? IsOpen is already marked as obsolete, so this is not necessary...

Comment (bigworld12) [inline MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs]: i use IsOpen to check if the dialog is visible so i close it programmatically , so it filled vs with warnings of it being obsolete

File: MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs [hunk 2]
```diff
@@ -12,23 +12,36 @@ public class ProgressDialogController
...
+   
+        /// <summary>
+        /// This event is raised when the associated <see cref="ProgressDialog"/> was cancelled by the user.
+        /// </summary>
+        public event EventHandler Cancelled;
+
+        /// <summary>
+        /// Gets if the Cancel button has been pressed.
+        /// </summary>        
+        public bool IsCancelled {get{ return m_iscanceled;}}
```
Comment (punker76) [inline MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs]: this can be simplyfied to `public bool IsCancelled { get; private set; }`
and i would prefere `IsCanceled`

Comment (bigworld12) [inline MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs]: ok i will bring it back

File: MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs [hunk 3]
```diff
@@ -12,16 +12,27 @@ public class ProgressDialogController
...
         /// <summary>
-        /// This event is raised when the associated <see cref="ProgressDialog"/> was closed.
+        /// This event is raised when the associated <see cref="ProgressDialog"/> was closed programmatically.
         /// </summary>
         public event EventHandler Closed;
+   
+        /// <summary>
+        /// This event is raised when the associated <see cref="ProgressDialog"/> was cancelled by the user.
+        /// </summary>
+        public event EventHandler Cancelled;
```
Comment (punker76) [inline MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs]: ;-) -> `Canceled`

Comment (remcoros) [inline MahApps.Metro/Controls/Dialogs/ProgressDialogController.cs]: Typical Americanism. Everywhere else in the world it's cancelled (with double l).

http://grammarist.com/spelling/cancel/
